producer.go

```diff
@@ -18,10 +18,11 @@ type Writer interface {
 }
 
 type producer struct {
-	w Writer
+	w           Writer
+	interceptor *ProducerInterceptor
```

Comment:
It is not necessary to use interface pointer. I mean `ProducerInterceptor` is interface, no need to use *